Formula/j/jsontoolkit.rb

```diff
@@ -0,0 +1,52 @@
+class Jsontoolkit < Formula
+  desc "Swiss-army knife library for expressive JSON programming in modern C++"
+  homepage "https://jsontoolkit.sourcemeta.com/"
+  url "https://github.com/sourcemeta/jsontoolkit/archive/refs/tags/v2.0.0.tar.gz"
+  sha256 "00f82f02166beabec80522e2bbc7b839ee9b7ccb631411c42e6fab65186e80ba"
+  license "AGPL-3.0-only"
+
+  depends_on "cmake" => :build
+
+  def install
+    args = %W[
+      -DCMAKE_COMPILE_WARNING_AS_ERROR=OFF
+      -DJSONTOOLKIT_CONTRIB=OFF
+      -DJSONTOOLKIT_TESTS=OFF
+      -DJSONTOOLKIT_DOCS=OFF
+      -DBUILD_SHARED_LIBS=ON
+      -DCMAKE_INSTALL_RPATH=#{rpath}
+    ]
+
+    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    (testpath/"test.cpp").write <<~EOS
+      #include <sourcemeta/jsontoolkit/json.h>
+      #include <sstream>
+      #include <iostream>
+      #include <cassert>
+      int main() {
+        std::istringstream input{"true"};
+        const sourcemeta::jsontoolkit::JSON document =
+          sourcemeta::jsontoolkit::parse(input);
+        assert(document.is_boolean());
+        assert(document.to_boolean());
+        std::cout << "JSON Toolkit works!" << std::endl;
+        return 0;
+      }
+    EOS
+    system ENV.cxx, "test.cpp", "-std=c++20", "-I#{include}",
+           "-L#{lib}",
+           "-Wl,-rpath,#{lib}",
+           "-lsourcemeta_jsontoolkit_json",
+           "-lsourcemeta_jsontoolkit_jsonschema",
+           "-lsourcemeta_jsontoolkit_jsonl",
+           "-lsourcemeta_jsontoolkit_jsonpointer",
+           "-lsourcemeta_jsontoolkit_uri",
```

Comment:
this can be put into pc file?

Reply:
@tony-go Actually, note that JSON Toolkit offers full blown `find_package` CMake integration. Instead of adding stuff manually, maybe create a test `CMakeLists.txt` that pulls it in? Like we do in this test one: https://github.com/sourcemeta/jsontoolkit/blob/main/test/packaging/find_package/CMakeLists.txt#L6

Reply:
Also, your test example is only using the JSON module, so you don't really need to link against the URI, JSON Schema, JSONL, and JSON Pointer modules.

Reply:
Yeah good idea @chenrui333 @jviotti :) let's create a cmake file. I tried to inline things in first place, probably because I found examples with that approach.

> Also, your test example is only using the JSON module, so you don't really need to link against the URI, JSON Schema, JSONL, and JSON Pointer modules.

Yeah, my goal here was to be sure that I could link to every shared libs. I would rather add a quick test for each of these.

Reply:
Ah, makes sense!

Reply:
yeah, there are examples on the cmake test build as well, you can check them out (it would also help people adopt this formula too in practice)

Reply:
hey @chenrui333 :) 

> there are examples on the cmake test build as well

You mean this: https://github.com/Homebrew/homebrew-core/blob/master/Formula/c/cmake.rb#L74 ?